src/pages/people-page/people.page.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../../components/Loader';
+import { getPeople } from '../../api';
+import { Person } from './../../types/Person';
+import { PersonLink } from '../person-link/person.link';
+import { useParams } from 'react-router-dom';
+
+export default function PeoplePage() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState<string | null>(
+    null,
+  );
+  const [error, setError] = useState<string | null>(null);
+  const { slug: targetPersonSlug } = useParams<{ slug: string }>();
+
+  useEffect(() => {
+    getPeople()
+      .then(fetchedPeople => {
+        setPeople(fetchedPeople);
+        setIsLoading(false);
+        setError(null);
+      })
+      .catch(() => {
+        setIsLoading(false);
+        setError('Something went wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handlePersonClick = (slug: string | null) => {
+    setSelectedPersonSlug(slug);
+  };
+
+  const shouldApplyBackground = (slug: string) =>
+    targetPersonSlug === slug ? 'has-background-warning' : '';
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      <div className="block">
+        <div className="box table-container">
+          {isLoading ? (
+            <Loader />
+          ) : error ? (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          ) : people.length === 0 ? (
+            <p data-cy="noPeopleMessage">There are no people on the server</p>
+          ) : (
+            <table
+              data-cy="peopleTable"
+              className="table is-striped is-hoverable is-narrow is-fullwidth"
+            >
+              <thead>
+                <tr>
+                  <th>Name</th>
+                  <th>Sex</th>
+                  <th>Born</th>
+                  <th>Died</th>
+                  <th>Mother</th>
+                  <th>Father</th>
+                </tr>
+              </thead>
+              <tbody>
+                {people.map(person => (
+                  <tr
+                    key={person.slug}
+                    data-cy="person"
+                    className={shouldApplyBackground(person.slug)}
+                  >
+                    <td className={shouldApplyBackground(person.slug)}>
+                      <PersonLink
+                        person={person}
+                        onClick={() => handlePersonClick(person.slug)}
+                      >
+                        {person.name}
+                      </PersonLink>
+                    </td>
+                    <td className={shouldApplyBackground(person.slug)}>
```

Comment:
This code looks way too repetitive, there must be a way to make it simpler...
What happens if all _td_ used a _PersonLink_ inside? Maybe you could use a _map_:
```
const tableCells = ['name', 'sex', 'born', 'died', 'motherName', 'fatherName'];

return (
...
{people.map(person => (
    <tr
        key={person.slug}
        data-cy="person"
        className={shouldApplyBackground(person.slug)}
    >
        {tableCells.map(cell => (
            <td>
                <PersonLink person={peopleMap[person[cell]]}>
                   {person[cell] !== undefined ? person[cell] : '-'}
               </PersonLink>
           </td>
......
```